pretixprint/app/src/main/java/eu/pretix/pretixprint/ui/Settings.kt

```diff
@@ -197,6 +209,33 @@ class SettingsFragment : PreferenceFragmentCompat() {
             startActivity(intent)
         }
     }
+
+    fun confirmRemovePrinter(type: String) {
+        pinProtect {
+            val typeRef = resources.getIdentifier("settings_label_${type}printer", "string", requireActivity().packageName)
+            val summary = printerSummary(type)
+            val alert = MaterialAlertDialogBuilder(requireActivity())
+                .setTitle(getString(R.string.pref_delete_printer, getString(typeRef)))
+                .setMessage(summary)
+                .setPositiveButton(R.string.action_delete) { dialog, _ ->
+                    removePrinter(type)
+                    onResume() // refresh preferences
+                    dialog.dismiss()
+                }
+                .setNegativeButton(android.R.string.cancel, null)
+                .create()
+            alert.show()
+        }
+    }
+
+    @SuppressLint("ApplySharedPref")
+    fun removePrinter(type: String) {
+        defaultSharedPreferences.edit()
+            .remove("hardware_${type}printer_ip")
+            .remove("hardware_${type}printer_printername")
+            .remove("hardware_${type}printer_connection")
+            .commit()
```

Comment:
Is there any downside with apply() here?

Reply:
ah right, fixed it :)